src/os/color.js

```diff
@@ -781,9 +781,12 @@ os.color.adjustColor = function(data, brightness, contrast, saturation) {
     var g = data[i + 1];
     var b = data[i + 2];
     var a = data[i + 3];
-    data[i] = (r * m[0] + g * m[5] + b * m[10] + a * m[15] + m[20]);
-    data[i + 1] = (r * m[1] + g * m[6] + b * m[11] + a * m[16] + m[21]);
-    data[i + 2] = (r * m[2] + g * m[7] + b * m[12] + a * m[17] + m[22]);
+    data[i] = Math.round((r * m[0] + g * m[5] + b * m[10] + a * m[15] + m[20]));
+    data[i + 1] = Math.round((r * m[1] + g * m[6] + b * m[11] + a * m[16] + m[21]));
+    data[i + 2] = Math.round((r * m[2] + g * m[7] + b * m[12] + a * m[17] + m[22]));
+    goog.math.clamp(data[i], 0, 255);
```

Comment:
Clamp doesn't modify the value in-place, it returns the clamped value. You'll want to do the clamp on the previous lines.